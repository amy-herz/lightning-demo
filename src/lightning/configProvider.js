import { getDefaultConfig } from './defaultConfig';

// Closure to hold the APIs if and when available
let PROVIDED_IMPL = getDefaultConfig();
let FORM_FACTOR;

let configProvided = false;

export default function configProviderService(serviceAPI) {
  if (!configProvided) {
    PROVIDED_IMPL = {
      getCoreInfo:
        serviceAPI.getInitializer &&
        serviceAPI.getInitializer('coreInfoConfig'),
      getPathPrefix: serviceAPI.getPathPrefix,
      getFormFactor: serviceAPI.getFormFactor,
      getToken: serviceAPI.getToken,
      getLocale: serviceAPI.getLocale,
      getLocalizationService: serviceAPI.getLocalizationService,
      sanitizeDOM: serviceAPI.sanitizeDOM
    };
    configProvided = true;
    FORM_FACTOR = undefined;
  } else {
    throw new Error(
      'ConfigProvider can only be set once at initilization time'
    );
  }

  return { name: 'lightning-config-provider' };
}

export function getPathPrefix() {
  return (
    (PROVIDED_IMPL &&
      PROVIDED_IMPL.getPathPrefix &&
      PROVIDED_IMPL.getPathPrefix()) ||
    ''
  );
}

export function getToken(name) {
  return (
    PROVIDED_IMPL && PROVIDED_IMPL.getToken && PROVIDED_IMPL.getToken(name)
  );
}

export function getLocale() {
  return PROVIDED_IMPL && PROVIDED_IMPL.getLocale && PROVIDED_IMPL.getLocale();
}

export function getFormFactor() {
  if (!FORM_FACTOR) {
    if (PROVIDED_IMPL && PROVIDED_IMPL.getFormFactor) {
      FORM_FACTOR = PROVIDED_IMPL.getFormFactor();
    } else {
      FORM_FACTOR = 'DESKTOP';
    }
  }

  return FORM_FACTOR;
}

export function getLocalizationService() {
  return (
    PROVIDED_IMPL &&
    PROVIDED_IMPL.getLocalizationService &&
    PROVIDED_IMPL.getLocalizationService()
  );
}

export function sanitizeDOM(dirty, config) {
  if (PROVIDED_IMPL && PROVIDED_IMPL.sanitizeDOM) {
    return PROVIDED_IMPL.sanitizeDOM(dirty, config);
  }
  return '';
}

export function getCoreInfo() {
  return (
    (PROVIDED_IMPL && PROVIDED_IMPL.getCoreInfo) || {
      untrustedContentDomain: '.a.forceusercontent.com',
      localhostPort: '',
      securePort: ''
    }
  );
}
