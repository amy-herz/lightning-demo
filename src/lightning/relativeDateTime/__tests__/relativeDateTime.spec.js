import { createElement } from 'lwc';
import Element from 'lightning/relativeDateTime';
// eslint-disable-next-line lwc/no-aura-libs
import { relativeFormat } from 'lightning:IntlLibrary';

const createRelativeDateTime = value => {
    const element = createElement('lightning-relative-date-time', {
        is: Element,
    });
    element.value = value;
    document.body.appendChild(element);
    return element;
};

beforeEach(() => {
    const format = relativeFormat().format;
    const validDate = Date.now();

    jest.useFakeTimers();

    while (format(validDate) !== 'hours') {} // eslint-disable-line no-empty
    format.mockClear();
});

describe('lightning-relative-date-time', () => {
    it('should throw when invalid value provided', () => {
        const elem = createRelativeDateTime();
        expect(() => {
            elem.value = 'not valid value';
        }).toThrow();
    });

    it('should change formattedValue with time', () => {
        jest.clearAllTimers();
        const elem = createRelativeDateTime(Date.now());
        const assertFormattedValueChanged = () => {
            expect(elem).toMatchSnapshot();
            jest.runOnlyPendingTimers();

            return Promise.resolve();
        };

        return Promise.resolve()
            .then(assertFormattedValueChanged)
            .then(assertFormattedValueChanged)
            .then(assertFormattedValueChanged);
    });
});
