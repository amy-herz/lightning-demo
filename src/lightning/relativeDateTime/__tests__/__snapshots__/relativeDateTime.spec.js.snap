// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lightning-relative-date-time should change formattedValue with time 1`] = `
<lightning-relative-date-time>
  #shadow-root(open)
    seconds
</lightning-relative-date-time>
`;

exports[`lightning-relative-date-time should change formattedValue with time 2`] = `
<lightning-relative-date-time>
  #shadow-root(open)
    minutes
</lightning-relative-date-time>
`;

exports[`lightning-relative-date-time should change formattedValue with time 3`] = `
<lightning-relative-date-time>
  #shadow-root(open)
    hours
</lightning-relative-date-time>
`;
